river/View.zig

```diff
@@ -427,7 +427,7 @@ pub fn setPendingOutput(view: *Self, output: *Output) void {
     if (view.pending.fullscreen) {
         view.pending.box = .{ .x = 0, .y = 0, .width = undefined, .height = undefined };
         output.wlr_output.effectiveResolution(&view.pending.box.width, &view.pending.box.height);
-    } else if (view.pending.float) {
+    } else {
```

Comment:
This change seems wrong to me. I think setting a position or dimension rule should force the view to start floating. We do need to document that this overrides `no-float` rules for example.

Reply:
I followed the recommendation given by @Leon-Plickat in https://github.com/riverwm/river/issues/865#issuecomment-1655499603. I thought that it would be a good idea to keep the float, position and dimensions rule independent for composibility. My interpretation of the position/dimensions rule is that it sets the position/dimensions of the view. It is up to the "mode" (not sure what the correct term is) to decide how the value gets used. However, I do see your point that really the only case where a position or dimensions rule would be used is for floating views.

Reply:
Hmm, yeah I guess we can go that way as well. In that case position and dimensions rules would have no effect unless the view is also matched by a float rule. They also have no effect if a view is started fullscreen, which seems reasonable. It would probably be a good idea to document this explicitly for clarity.

And, unless I'm missing something, it's still fine to revert this part of the patch. If views are not started floating then their position/dimensions will be overridden by the layout generator before they are sent to the client.

Reply:
> If views are not started floating then their position/dimensions will be overridden by the layout generator before they are sent to the client.

Ah, I didn't know that was the case. I reverted the change and documented the interaction of position and dimensions rules with float and fullscreen modes.